packages/components/progress/src/progress.vue

```diff
@@ -87,177 +87,148 @@ import {
+})
 
-    const trailPathStyle = computed(
-      (): CSSProperties => ({
-        strokeDasharray: `${perimeter.value * rate.value}px, ${
-          perimeter.value
-        }px`,
-        strokeDashoffset: strokeDashoffset.value,
-      })
-    )
+const perimeter = computed(() => 2 * Math.PI * radius.value)
 
-    const circlePathStyle = computed(
-      (): CSSProperties => ({
-        strokeDasharray: `${
-          perimeter.value * rate.value * (props.percentage / 100)
-        }px, ${perimeter.value}px`,
-        strokeDashoffset: strokeDashoffset.value,
-        transition:
-          'stroke-dasharray 0.6s ease 0s, stroke 0.6s ease, opacity ease 0.6s',
-      })
-    )
+const rate = computed(() => (props.type === 'dashboard' ? 0.75 : 1))
 
-    const stroke = computed(() => {
-      let ret: string
-      if (props.color) {
-        ret = getCurrentColor(props.percentage)
-      } else {
-        switch (props.status) {
-          case 'success':
-            ret = '#13ce66'
-            break
-          case 'exception':
-            ret = '#ff4949'
-            break
-          case 'warning':
-            ret = '#e6a23c'
-            break
-          default:
-            ret = '#20a0ff'
-        }
-      }
-      return ret
-    })
-
-    const statusIcon = computed(() => {
-      if (props.status === 'warning') {
-        return WarningFilled
-      }
-      if (props.type === 'line') {
-        return props.status === 'success' ? CircleCheck : CircleClose
-      } else {
-        return props.status === 'success' ? Check : Close
-      }
-    })
+const strokeDashoffset = computed(() => {
+  const offset = (-1 * perimeter.value * (1 - rate.value)) / 2
+  return `${offset}px`
+})
 
-    const progressTextSize = computed(() => {
-      return props.type === 'line'
-        ? 12 + props.strokeWidth * 0.4
-        : props.width * 0.111111 + 2
-    })
+const trailPathStyle = computed(
+  (): CSSProperties => ({
+    strokeDasharray: `${perimeter.value * rate.value}px, ${perimeter.value}px`,
+    strokeDashoffset: strokeDashoffset.value,
+  })
+)
+
+const circlePathStyle = computed(
+  (): CSSProperties => ({
+    strokeDasharray: `${
+      perimeter.value * rate.value * (props.percentage / 100)
+    }px, ${perimeter.value}px`,
+    strokeDashoffset: strokeDashoffset.value,
+    transition:
+      'stroke-dasharray 0.6s ease 0s, stroke 0.6s ease, opacity ease 0.6s',
+  })
+)
+
+const stroke = computed(() => {
+  let ret: string
+  if (props.color) {
+    ret = getCurrentColor(props.percentage)
+  } else {
+    ret = COLOR_STATUS_MAP[props.status] || COLOR_STATUS_MAP.default
+  }
+  return ret
+})
 
-    const content = computed(() => props.format(props.percentage))
+const statusIcon = computed(() => {
+  if (props.status === 'warning') {
+    return WarningFilled
+  }
+  if (props.type === 'line') {
+    return props.status === 'success' ? CircleCheck : CircleClose
+  } else {
+    return props.status === 'success' ? Check : Close
+  }
```

Comment:
```suggestion
  switch(props.status) {
    case 'warning':
      return WarningFilled
    case 'line':
      return props.status === 'success' ? CircleCheck : CircleClose
    default:
      return props.status === 'success' ? Check : Close
  }
```

Reply:
`line` is a type, if `const type = props.status === 'warning' ? 'warning' : props.type` This seems to be an extra step